THREADING.md

```diff
@@ -100,6 +104,13 @@ direct connection to libgcrypt and thus has not convenience functions for
 it (but libgcrypt has macros). Read libgcrypt's
 [threading documentation for more information](http://www.gnupg.org/documentation/manuals/gcrypt/Multi_002dThreading.html)
 
+libssh2 provides a `libssh2_init` function, which is called
+automatically by libssh2 code if needed.  However, `libssh2_init`
+is not thread-safe, and the way in which it is automatically called
+is also not thread-safe, so in a multi-threaded environment you may need
+to call libssh2_init directly to prevent the thread-unsafe automatic
+initialization.
```

Comment:
This isn't about thread safety, but about initializing our dependencies. Locking is either automatic or a *application-wide* configuration so libgit2 can't assume it knows what to do.